src/pools/D3M4626TypePool.sol

```diff
@@ -0,0 +1,163 @@
+    /* EVENTS */
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+
+    /* CONSTANTS */
+
+    IERC20 public immutable dai;
+    IERC4626 public immutable vault;
+    VatLike public immutable vat;
+
+    /* STORAGE */
+
+    address public hub;
+    mapping(address => bool) public wards;
+
+    /* CONSTRUCTOR */
+
+    constructor(address newDai, address newVault, address newVat) {
+        require(newDai != address(0), "D3M4626TypePool/zero-address");
+        require(newVat != address(0), "D3M4626TypePool/zero-address");
+        // Implicitely checks that vault addres is not zero.
+        require(IERC4626(newVault).asset() == address(newDai), "D3M4626TypePool/vault-asset-is-not-dai");
+
+        dai = IERC20(newDai);
+        vault = IERC4626(newVault);
+        vat = VatLike(newVat);
+
+        wards[msg.sender] = true;
+        emit Rely(msg.sender);
+
+        dai.approve(address(vault), type(uint256).max);
+    }
+
+    /* MODIFIERS */
+
+    modifier onlyHub() {
+        require(msg.sender == hub, "D3M4626TypePool/only-hub");
+        _;
+    }
+
+    modifier onlyAuthorized() {
+        require(wards[msg.sender], "D3M4626TypePool/not-authorized");
+        _;
+    }
+
+    /* ONLY HUB */
+
+    /// @inheritdoc ID3MPool
+    function deposit(uint256 assets) external override onlyHub {
+        vault.deposit(assets, address(this));
+    }
+
+    /// @inheritdoc ID3MPool
+    function withdraw(uint256 assets) external override onlyHub {
+        vault.withdraw(assets, msg.sender, address(this));
+    }
+
+    /// @inheritdoc ID3MPool
+    function exit(address dst, uint256 shares) external onlyHub {
+        vault.transfer(dst, shares);
+    }
+
+    /* ONLY AUTHORIZED */
+
+    /// @inheritdoc ID3MPool
+    function quit(address dst) external onlyAuthorized {
+        require(vat.live() == 1, "D3M4626TypePool/no-quit-during-shutdown");
+        
+        vault.transfer(dst, vault.balanceOf(address(this)));
+    }
+
+    function rely(address usr) public onlyAuthorized {
+        wards[usr] = true;
+        emit Rely(usr);
+    }
+
+    function deny(address usr) external onlyAuthorized {
+        wards[usr] = false;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, address data) external onlyAuthorized {
+        require(vat.live() == 1, "D3M4626TypePool/no-file-during-shutdown");
+        require(what == "hub", "D3M4626TypePool/file-unrecognized-param");
+
+        vat.nope(hub);
+        hub = data;
+        vat.hope(data);
+
+        emit File(what, data);
+    }
+
+    function urdClaim(address dst, address distributor, address reward, uint256 claimable, bytes32[] calldata proof) external payable {
+        require(dst != address(0), "D3M4626TypePool/zero-address");
+
+        uint256 amount = IUniversalRewardsDistributor(distributor).claim(address(this), reward, claimable, proof);
+        IERC20(reward).transfer(dst, amount);
```

Comment:
This seems like users can permissionlessly steal the rewards of the d3m. All rewards should just accrue to king address same as the other pools: https://github.com/makerdao/dss-direct-deposit/blob/master/src/pools/D3MAaveV3NoSupplyCapTypePool.sol#L247

Reply:
Actually in the interest of simplicity let's just remove the reward collection for now. We can upgrade the pool later if need be.

Reply:
actually rewards are claimable by anybody (and will be transferred to the entitled account), so we could just have a onlyAuth skim function